utils/roll_browser.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2017 Google Inc. All rights reserved.
+ * Modifications copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const path = require('path');
+const fs = require('fs');
+const protocolGenerator = require('./protocol-types-generator');
+
+const SCRIPT_NAME = path.basename(__filename);
+const ROOT_PATH = path.resolve(path.join(__dirname, '..'));
+
+function usage() {
+  return `
+usage: ${SCRIPT_NAME} <browser> <revision>
+
+Roll the <browser> to a specific <revision> and generate new protocol.
+Supported browsers: chromium, firefox, webkit.
+
+Example:
+  ${SCRIPT_NAME} chromium 123456
+`;
+}
+
+(async () => {
+  // 1. Parse CLI arguments
+  const args = process.argv.slice(2);
+  if (args.some(arg => arg === '--help')) {
+    console.log(usage());
+    process.exit(1);
+  } else if (args.length < 1) {
+    console.log(`Please specify the browser name, e.g. 'chromium'.`);
+    console.log(`Try running ${SCRIPT_NAME} --help`);
+    process.exit(1);
+  } else if (args.length < 2) {
+    console.log(`Please specify the revision`);
+    console.log(`Try running ${SCRIPT_NAME} --help`);
+    process.exit(1);
+  }
+  const browserName = args[0];
+  if (!['chromium', 'firefox', 'webkit'].includes(browserName)) {
+    console.log(`Unknown browser "${browserName}"`);
+    console.log(`Try running ${SCRIPT_NAME} --help`);
+    process.exit(1);
+  }
+  const revision = args[1];
+
+  // 2. Update browsers.json.
```

Comment:
can we switch step 2 and step 3 so that it won't write browsers.json unless it successfully downloaded the browser?